liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,401 @@
+
+      // Adjust for native decimals
+      entry.price = new BigNumber(price.toString()).dividedBy(oneInchMult);
+
+      // Update the Collateral Factor
+      const market = await comptrollerContract.markets(currentToken);
+      // Per Compound: https://compound.finance/docs/comptroller#collateral-factor
+      //   "collateralFactorMantissa, scaled by 1e18, is multiplied by a supply balance to determine
+      //    how much value can be borrowed"
+      entry.collateralMult = new BigNumber(market[1].toString())
+        .dividedBy(BigNumber(10).pow(18));
+    }));
+
+    // Calculate health on all accounts
+    Object.keys(accounts).forEach((account) => {
+      let borrowBalance = new BigNumber(0);
+      let supplyBalance = new BigNumber(0);
+      Object.entries(tokens).forEach(([token, entry]) => {
+        const {
+          exchangeRateMult,
+          price,
+          collateralMult,
+        } = entry;
+
+        // Ref: https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/Comptroller.sol#L751
+        const tokensToDenom = exchangeRateMult.times(price.times(collateralMult));
+
+        if (supply[token][account]) {
+          // Supply balances are stored in cTokens and need to be multiplied by the
+          //   exchange rate, price and the collateral factor to determine the ETH value
+          //   of the collateral.
+          // Ref: https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/Comptroller.sol#L754
+          supplyBalance = supplyBalance.plus(supply[token][account].times(tokensToDenom));
+        }
+
+        // Ref: https://github.com/compound-finance/compound-protocol/blob/3affca87636eecd901eb43f81a4813186393905d/contracts/Comptroller.sol#L757
+        if (borrow[token][account]) {
+          // Only need to multiply by the price.
+          borrowBalance = borrowBalance.plus(borrow[token][account].times(price));
+        }
+      });
+
+      // Remove non-borrowers
+      if (borrowBalance.eq(0)) {
+        delete accounts[account];
+      } else {
+        accounts[account].supplyBalance = supplyBalance;
+        accounts[account].borrowBalance = borrowBalance;
+        accounts[account].health = supplyBalance.dividedBy(borrowBalance);
+      }
+    });
+
+    // Check the Comptroller contract for actual liquidity if the health is below x
+    const lowHealthAccounts = [];
+    Object.keys(accounts).forEach((currentAccount) => {
+      if (accounts[currentAccount].health.isLessThan(data.lowHealthThreshold)) {
+        lowHealthAccounts.push(currentAccount);
+      }
+    });
+    await Promise.all(lowHealthAccounts.map(async (currentAccount) => {
+      const liquidity = await comptrollerContract.getAccountLiquidity(currentAccount);
+      const shortfallUSD = new BigNumber(ethers.utils.formatEther(liquidity[2]).toString());
```

Comment:
My guess is that the returned value in `liquidity[2]` will be represented as an `ethers.BigNumber`.  If that's true, then there _should_ be a `toString()` method available on it.
```suggestion
      const shortfallUSD = new BigNumber(liquidity[2].toString());
```

Reply:
`formatEther` was a quick call to convert 1e18 wei to 1 Ether. If I just do `toString()` then I'll additionally need to divide by 1e18 to get decimal numbers for outputs.

Reply:
Clarified in commit 17ed2a115215bd20ef5be24f23bd2a6ee856d910